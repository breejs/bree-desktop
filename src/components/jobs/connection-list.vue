<script setup>
import ConnectionListItem from './connection-list-item.vue';

const props = defineProps({
  connections: {
    type: Array,
    required: true
  }
});
</script>

<template lang="pug">
ul.list-group.list-group-flush
  template(v-if='props.connections?.length > 0')
    ConnectionListItem(
      v-for='connection in props.connections',
      :key='connection.name',
      :connection='connection'
    )

  template(v-else)
    li.list-group-item
      = 'No connections'
</template>
