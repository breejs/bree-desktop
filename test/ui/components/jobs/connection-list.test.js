import { storyTests } from '../../_story-tests';

import * as stories from '@/components/jobs/connection-list.stories';

storyTests(stories);
