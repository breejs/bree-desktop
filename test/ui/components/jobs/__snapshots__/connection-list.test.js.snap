// Vitest Snapshot v1

exports[`story snapshots > Connections 1`] = `
"<ul class=\\"list-group list-group-flush\\">
  <li class=\\"list-group-item\\">
    <div class=\\"row align-items-center\\">
      <div class=\\"col col-auto pointer\\"><i class=\\"bi bi-chevron-right\\"></i>
        <!--v-if-->
      </div>
      <div class=\\"col col-auto\\">
        <!--v-if--><i class=\\"bi fs-3 text-active bi-card-list\\" data-v-3b64800a=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Active\\" aria-label=\\"Active\\"></i>
      </div>
      <div class=\\"col text-start lh-1\\">
        <div class=\\"row\\">
          <div class=\\"col\\"><small>Running Connection
              <!--v-if-->
            </small></div>
        </div>
        <div class=\\"row\\">
          <div class=\\"col\\"><small class=\\"text-muted\\">Last Ping: 3 months ago</small></div>
        </div>
      </div>
      <div class=\\"col col-auto\\" style=\\"display: none;\\"><button class=\\"btn btn-outline-danger me-1\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Stop\\" aria-label=\\"Stop\\"><i class=\\"bi bi-stop-fill\\"></i></button>
        <!--v-if--><button class=\\"btn btn-outline-warning me-1\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Restart\\" aria-label=\\"Restart\\"><i class=\\"bi bi-arrow-clockwise\\"></i></button><!-- button.btn.btn-outline-danger(-->
        <!--   v-tooltip:title='\\"Delete\\"',-->
        <!--   @click='kind === \\"connection\\" ? remove(job.name) : $emit(\\"remove\\", job.name)'-->
        <!-- )-->
        <!--   i.bi.bi-trash-fill-->
      </div>
    </div>
    <!--v-if-->
  </li>
  <li class=\\"list-group-item\\">
    <div class=\\"row align-items-center\\">
      <div class=\\"col col-auto pointer\\"><i class=\\"bi bi-chevron-right\\"></i>
        <!--v-if-->
      </div>
      <div class=\\"col col-auto\\">
        <!--v-if--><i class=\\"bi fs-3 text-done bi-card-list\\" data-v-3b64800a=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Done\\" aria-label=\\"Done\\"></i>
      </div>
      <div class=\\"col text-start lh-1\\">
        <div class=\\"row\\">
          <div class=\\"col\\"><small>Stopped Connection
              <!--v-if-->
            </small></div>
        </div>
        <div class=\\"row\\">
          <div class=\\"col\\"><small class=\\"text-muted\\">Last Ping: 3 months ago</small></div>
        </div>
      </div>
      <div class=\\"col col-auto\\" style=\\"display: none;\\">
        <!--v-if--><button class=\\"btn btn-outline-success me-1\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Start\\" aria-label=\\"Start\\"><i class=\\"bi bi-play-fill\\"></i></button><button class=\\"btn btn-outline-warning me-1\\" disabled=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Restart\\" aria-label=\\"Restart\\"><i class=\\"bi bi-arrow-clockwise\\"></i></button><!-- button.btn.btn-outline-danger(-->
        <!--   v-tooltip:title='\\"Delete\\"',-->
        <!--   @click='kind === \\"connection\\" ? remove(job.name) : $emit(\\"remove\\", job.name)'-->
        <!-- )-->
        <!--   i.bi.bi-trash-fill-->
      </div>
    </div>
    <!--v-if-->
  </li>
</ul>"
`;

exports[`story snapshots > NoConnections 1`] = `
"<ul class=\\"list-group list-group-flush\\">
  <li class=\\"list-group-item\\">No connections</li>
</ul>"
`;
